include/aws/common/byte_buf.h

```diff
@@ -438,85 +455,103 @@ static inline bool aws_byte_cursor_read_be64(struct aws_byte_cursor *cur, uint64
 }
 
 /**
- * Write specified number of bytes from array to byte cursor.
+ * Write specified number of bytes from array to byte buffer.
  *
- * On success, returns true and updates the cursor pointer/length accordingly.
- * If there is insufficient space in the cursor, returns false, leaving the
- * cursor unchanged.
+ * On success, returns true and updates the buffer length accordingly.
+ * If there is insufficient space in the buffer, returns false, leaving the
+ * buffer unchanged.
  */
-static inline bool aws_byte_cursor_write(
-    struct aws_byte_cursor *AWS_RESTRICT cur,
+static inline bool aws_byte_buf_write(
+    struct aws_byte_buf *AWS_RESTRICT cur,
     const uint8_t *AWS_RESTRICT src,
     size_t len) {
-    struct aws_byte_cursor slice = aws_byte_cursor_advance_nospec(cur, len);
+
+    struct aws_byte_cursor write_cursor;
+    write_cursor.ptr = cur->buffer + cur->len;
+    write_cursor.len = cur->capacity - cur->len;
+    struct aws_byte_cursor slice = aws_byte_cursor_advance(&write_cursor, len);
 
     if (slice.ptr) {
         memcpy(slice.ptr, src, len);
+        cur->len += len;
         return true;
     }
     return false;
 }
 
 /**
- * Copies all bytes from buffer to cursor.
+ * Copies all bytes from buffer to buffer.
  *
- * On success, returns true and updates the cursor pointer/length accordingly.
- * If there is insufficient space in the cursor, returns false, leaving the
- * cursor unchanged.
+ * On success, returns true and updates the buffer /length accordingly.
+ * If there is insufficient space in the buffer, returns false, leaving the
+ * buffer unchanged.
  */
-static inline bool aws_byte_cursor_write_from_whole_buffer(
-    struct aws_byte_cursor *AWS_RESTRICT cur,
+static inline bool aws_byte_buf_write_from_whole_buffer(
+    struct aws_byte_buf *AWS_RESTRICT cur,
     const struct aws_byte_buf *AWS_RESTRICT src) {
-    return aws_byte_cursor_write(cur, src->buffer, src->len);
+    return aws_byte_buf_write(cur, src->buffer, src->len);
 }
 
 /**
- * Copies one byte to cursor.
+ * Copies all bytes from buffer to buffer.
  *
- * On success, returns true and updates the cursor pointer/length
- accordingly.<<<<<< str-split
+ * On success, returns true and updates the buffer /length accordingly.
+ * If there is insufficient space in the buffer, returns false, leaving the
+ * buffer unchanged.
+ */
+static inline bool aws_byte_buf_write_from_whole_cursor(
+    struct aws_byte_buf *AWS_RESTRICT cur,
+    const struct aws_byte_cursor *AWS_RESTRICT src) {
+    return aws_byte_buf_write(cur, src->ptr, src->len);
```

Comment:
I don't think so, since it would empty the cursor out. I'd opt to not clear potentially useful information in favor of letting the user do that if they want to. Especially since this would work pretty well:
```c
aws_byte_buf_write_from_whole_cursor(out_buf,
        aws_byte_cursor_advance(my_data_cursor, num_bytes_to_write));
```